### PR TITLE
[RFC] Move RADOS support to separate repo

### PR DESCRIPTION
**Pull request by riversand963**
This PR moves RADOS support from RocksDB repo to a separate repo. The new (temporary?) repo
in this PR serves as an example before we finalize the decision on where and who to host RADOS support. At this point,
people can start from the example repo and fork.

The goal is to include this commit in RocksDB 7.0 release.

Reference:
https://github.com/ajkr/dedupfs by @ajkr

Test plan:
Follow instructions in https://github.com/riversand963/rocksdb-rados-env/blob/main/README.md and build
test binary `env_librados_test` and run it.

Also, make check

### CODE REVIEW

File: PLUGINS.md [hunk 1]
```diff
@@ -2,3 +2,4 @@ This is the list of all known third-party plugins for RocksDB. If something is m
 
 * [Dedupfs](https://github.com/ajkr/dedupfs): an example for plugin developers to reference
 * [ZenFS](https://github.com/westerndigitalcorporation/zenfs): a file system for zoned block devices
+* [RADOS](https://github.com/riversand963/rocksdb-hdfs-env): an Env used for interacting with RADOS. Migrated from RocksDB main repo.
```
Comment (ajkr) [inline PLUGINS.md:5]: Can you double check the URL? The RADOS item has "rocksdb-hdfs-env" in it.

**Review by ajkr — APPROVED** [review]
LGTM

File: PLUGINS.md (hunk 1 above)
Comment (riversand963) [reply PLUGINS.md]: Fixed.

**Comment by riversand963** [comment]
Thanks @ajkr for the review!

**Pull request closed without merging**